src/main.cpp

```diff
@@ -575,7 +574,178 @@ void affect_nearby_friends_with_alarm(struct Thing *traptng)
+    }
+    else if ( blocked_flags & SlbBloF_WalledY )
+    {
+      angle = thing->move_angle_xy;
+      if ( (angle <= 0x200u) || (angle > 0x600u) )
+        collide = process_boulder_collision(thing, &pos, 0, -1);
+      else
+        collide = process_boulder_collision(thing, &pos, 0, 1);
+    }
+    else
+    {
+      collide = 0;
+    }
+    if ( collide != 1 )
+    {
+      if ( (thing->model != ShM_SolidBoulder) && (collide == 0) )
+      {
+        thing->health -= game.boulder_reduce_health_wall;
+      }
+      slide_thing_against_wall_at(thing, &pos, blocked_flags);
+      if ( blocked_flags & SlbBloF_WalledX )
+      {
+        angle = thing->move_angle_xy;
+        if ( (angle) && ( (angle <= 0x200u) || (angle > 0x600u) ) )
+        {
+          unsigned short y = *(unsigned short *)&thing->mappos.y.stl;
+          *(unsigned short *)&pos2.x.stl = *(unsigned short *)&thing->mappos.x.stl;
+          *(unsigned short *)&pos2.z.stl = 0;
+          *(unsigned short *)&pos2.y.stl = y - 3 * speed;
+          *(unsigned short *)&pos2.z.stl = get_thing_height_at(thing, &pos2);
+          new_angle = (thing_in_wall_at(thing, &pos2) < 1) ? 0 : 0x400;
+        }
+        else
+        {
+          *(unsigned short *)&pos2.x.stl = *(unsigned short *)&thing->mappos.x.stl;
+          *(unsigned short *)&pos2.z.stl = 0;
+          *(unsigned short *)&pos2.y.stl = *(unsigned short *)&thing->mappos.y.stl + 3 * speed;
+          *(unsigned short *)&pos2.z.stl = get_thing_height_at(thing, &pos2);
+          new_angle = (thing_in_wall_at(thing, &pos2) < 1) ? 0x400 : 0;
+        }
+      }
+      else if ( blocked_flags & SlbBloF_WalledY )
+      {
+        angle = thing->move_angle_xy;
+        if ( (angle) && (angle <= 0x400u) ) 
+        {
+          *(unsigned short *)&pos2.z.stl = 0;
+          *(unsigned short *)&pos2.y.stl = *(unsigned short *)&thing->mappos.y.stl;
+          *(unsigned short *)&pos2.x.stl = *(unsigned short *)&thing->mappos.x.stl + 3 * speed;
+          *(unsigned short *)&pos2.z.stl = get_thing_height_at(thing, &pos2);
+          new_angle = (thing_in_wall_at(thing, &pos2) < 1) ? 512 : 1536;
+        }
+        else
+        {
+          unsigned short x = *(unsigned short *)&thing->mappos.x.stl;
+          *(unsigned short *)&pos2.z.stl = 0;
+          *(unsigned short *)&pos2.y.stl = *(unsigned short *)&thing->mappos.y.stl;
+          *(unsigned short *)&pos2.x.stl = x - 3 * speed;
+          *(unsigned short *)&pos2.z.stl = get_thing_height_at(thing, &pos2);
+          new_angle = (thing_in_wall_at(thing, &pos2) < 1) ? 1536 : 512;
+        }
+      }
+      else
+      {
+        ERRORLOG("Cannot find boulder wall hug angle!");
+        new_angle = 0;
+      }
+      thing->move_angle_xy = new_angle;
+    }
+  }
+  angle = thing->move_angle_xy;
+  *(unsigned short *)&thing->velocity.x.stl = shotst->speed
+                                       * lbSinTable[angle] >> 16;
```

Comment:
There's a 'LbSinL' function, isn't that better to use here?

Reply:
I'm no expert, but it doesn't appear to be the same.

Reply:
the thing the LbSinL does over directly accessing the array is applying a bitmask so that values larger 2048 get rolled over and still falling into the array, so basically if the value < 2048 no difference, if > 2048 it it would put out garbage that happens to be placed next to the array so I's say Loobinex is correct and you should call that function instead of the array directly